TESTING.md

```diff
@@ -0,0 +1,417 @@
+
+```gql
+query {
+  getQuestionnaire(id: 1) {
+    questionnaire {
+      sections {
+        pages {
+          id
+
+          # inline fragment for `QuestionPage` type
+          ... on QuestionPage {
+            guidance
+            answers {
+              id
+              label
+            }
+          }
+
+          # For purposes of example only. `InterstitialPage` doesn't exist yet
+          ... on InterstitialPage {
+            # doesn't exist yet
+            someField
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+### Testing through GraphiQL
+
+There are [queries](tests/fixtures/queries.gql) and [example data](tests/fixtures/data.json) in the [fixtures folder](tests/fixtures). These can be used with graphiql to manually build up a questionnaire.
+
+### DB migrations
+
+First start app using Docker.
+
+#### Create migration
+
+```
+yarn knex migrate:make name_of_migration
+```
+
+Where `name_of_migration` is the name you wish to use. e.g. `create_questionnaires_table`
+
+#### Apply migrations
+
+```
+docker-compose exec web yarn knex migrate:latest
+```
+
+#### Rollback migrations
+
+```
+docker-compose exec web yarn knex migrate:rollback
+```
+
+### Tests
+
+`yarn test` will start a single run of unit and integration tests.
+
+`yarn test --watch` will start unit and integration tests in watch mode.
+
+### Debugging (with VS Code)
+
+#### Debugging app
+
+Follow [this guide](https://github.com/docker/labs/blob/83514855aff21eaed3925d1fd28091b23de0e147/developer-tools/nodejs-debugging/VSCode-README.md) to enable debugging through VS Code.
+
+Use this config for VS Code, rather than what is detailed in the guide. This will attach _to the running docker container_:
+
+```json
+{
+  "version": "0.2.0",
+  "configurations": [
+    {
+      "name": "Attach to Container",
+      "type": "node",
+      "request": "attach",
+      "port": 5858,
+      "address": "localhost",
+      "restart": true,
+      "sourceMaps": false,
+      "localRoot": "${workspaceRoot}",
+      "remoteRoot": "/app",
+      "protocol": "inspector"
+    }
+  ]
+}
+```
+
+#### Debugging tests
+
+Add the following to your `launch.json` configuration:
+
+```json
+{
+  "name": "Attach by Process ID",
+  "type": "node",
+  "request": "attach",
+  "processId": "${command:PickProcess}"
+}
+```
+
+Then start your tests [as described above](#tests). You can now start a debugging session, and pick the jest process to attach to.
+
+### Importing questionnaires
+
+There is a dev only endpoint exposed in the dev environment to be able to import questionnaires from other environments.
+
+#### How to use it
+
+1. Run the following query against the environment to retrieve the questionnaire. You need to provide the id as well. (You could use <https://github.com/skevy/graphiql-app>)
+
+```graphql
+fragment answerFragment on Answer {
```

Comment:
This will likely get out of date. Better to link to the file containing the query.